spec/state_of_the_nation_spec.rb

```diff
@@ -245,6 +245,97 @@ class Country < ActiveRecord::Base
+    end
+
+    it "returns false for records active outside the range" do
+      expect(bounded_president).not_to be_active_in_interval(day(4), day(7))
+      expect(bounded_president).not_to be_active_in_interval(day(4), nil)
+
+      expect(unbounded_president).not_to be_active_in_interval(day(2), day(4))
+      expect(unbounded_president).not_to be_active_in_interval(nil, day(4))
+
+      expect(president_with_empty_active_period).not_to be_active_in_interval(day(4), day(6))
+      expect(president_with_empty_active_period).not_to be_active_in_interval(day(4), nil)
+      expect(president_with_empty_active_period).not_to be_active_in_interval( nil, day(4))
+    end
+
+    context "with ignore_empty as true" do
+      before do
+        allow(President).to receive(:ignore_empty).and_return(true)
+      end
+
+      it "returns true for records that are active in the interval" do
+        expect(bounded_president).to be_active_in_interval(day(3), day(7))
+        expect(bounded_president).to be_active_in_interval(day(3), nil)
+        expect(bounded_president).to be_active_in_interval(nil, day(3))
+        expect(bounded_president).to be_active_in_interval(nil, nil)
+
+        expect(unbounded_president).to be_active_in_interval(day(4), day(12))
+        expect(unbounded_president).to be_active_in_interval(day(4), nil)
+        expect(unbounded_president).to be_active_in_interval(nil, day(12))
+        expect(unbounded_president).to be_active_in_interval(nil, nil)
+        expect(unbounded_president).to be_active_in_interval(day(4), day(4))
+      end
+
+      it "returns false for records with an empty activation period in the range" do
+        expect(president_with_empty_active_period).not_to be_active_in_interval(day(3), day(5))
+        expect(president_with_empty_active_period).not_to be_active_in_interval(day(3), nil)
+        expect(president_with_empty_active_period).not_to be_active_in_interval(nil, day(5))
+        expect(president_with_empty_active_period).not_to be_active_in_interval(nil, nil)
+      end
+
+      it "returns false for records active outside the range" do
+        expect(bounded_president).not_to be_active_in_interval(day(4), day(7))
+        expect(bounded_president).not_to be_active_in_interval(day(4), nil)
+
+        expect(unbounded_president).not_to be_active_in_interval(day(2), day(4))
+        expect(unbounded_president).not_to be_active_in_interval(nil, day(4))
+
+        expect(president_with_empty_active_period).not_to be_active_in_interval(day(4), day(6))
+        expect(president_with_empty_active_period).not_to be_active_in_interval(day(4), nil)
+        expect(president_with_empty_active_period).not_to be_active_in_interval( nil, day(4))
+      end
+
+    end
+
+    context "for records without a start date set" do
+      before do
+        travel_to(day(30))
+      end
+
+      let(:open_started_president) { country.presidents.create!(entered_office_at: nil, left_office_at: day(60)) }
```

Comment:
~Huh, why doesn't this fail on the `ensure_finishes_after_starts` validation? (specifically, on comparing `finish` to `nil` on line 125)~

Looks like it's because we use `start` and `finish` rather than the underlying values, and `start` will default to `now` (please make the default `now` stop, I want to get off Mr Bones' Wild Ride).